add test for shellscript syntax and quality

phlogistonjohn opened the pull request:
These changes:
* add a checker script
* make a bunch of largely mechanical changes suggested by shellcheck
* tag some files as not to be checked (for now)
* tag some lines to ignore a particular shellcheck rule

I decided not to try and fix some of the more complex files in this PR as there's already a lot to review here. I fully intend on removing the skip markers that are also marked fixme.

obnoxxx commented on tests/020-test-shellscripts.sh:
```diff
@@ -0,0 +1,46 @@
+fi
+
+cd "${BASE_DIR}"
+SCRIPTS=$(find -name '*.sh' | grep -v ./vendor | sort)
```
we need to also grep -v ./.git here.

phlogistonjohn commented:
Updated patch to exlude .git (using find itself, let me know if this approach is ok or not).
Also found and fixed two issues: one with the comment tag check causing the script to ignore itself and  an issue with flagging all files after a failed check as failed.

obnoxxx commented:
LGTM, thanks.
